### DISCUSSION TITLE
功能建议 玩家和ai打法不一样的时候 显示天凤的进张数统计

### DISCUSSION THREAD

**Discussion by chucklu** [Equim-chan/mjai-reviewer#124] [Ideas]
比如这个牌谱的东2局2本场
https://mjai.ekyu.moe/report/5dd6ac5b896a827a.html#kyoku-1-2
现在的分析只是简单地给出了玩家的出牌和ai的建议，
如下图显示玩家打出5s，ai建议打出9m。但是对于新手玩家来说，不是太理解这个。
![image](https://github.com/Equim-chan/mjai-reviewer/assets/3004057/abbbb68f-b442-4d4e-b00a-681dd220101b)

![image](https://github.com/Equim-chan/mjai-reviewer/assets/3004057/7cf04432-831f-4155-bd68-63e840ea16e4)

**Comment by chucklu** [comment]
我尝试在天凤网站查看两种打法(玩家的打法和ai的打法)的进张数统计，很明显就看出两种打法的差距。而且根据进张数的不同，可以很容易知道，差距在哪里。就这个例子而言，看到了进张的对比，我能理解，打出9m，前面就保留了一杯口的可能性。
https://tenhou.net/2/?q=445679m45579s33z9s
打9m 摸[4m5m6m7m3s4s5s6s7s8s9s3z 35枚]
打5s 摸[4m8m3s6s8s9s3z 22枚]

**Comment by chucklu** [comment]
打9m 摸[4m5m6m7m3s4s5s6s7s8s9s3z 35枚]
打5s 摸[4m8m3s6s8s9s3z 22枚]
不过这个要肉眼去看，两种打法的进张数的区别。

如果能添加显示进张的功能，同时告知用户两种打法的具体差异，就更好了。
比如这个例子
打9m 摸[4m5m6m7m3s4s5s6s7s8s9s3z 35枚] 
打5s 摸[4m8m3s6s8s9s3z 22枚]  

4m5m6m7m  多了567m，少了8m
4m8m 多了8m,但是少了567m

3s4s5s6s7s8s9s  多了457s
3s6s8s9s 少了457s，没有多任何进张。

能直接把这种多少的对比显示出来，就更人性化了。用户直接看这里的多少的分析，就大致能理解ai的想法

**Comment by chucklu** [comment]
上面的例子还不明显，发现另外一个例子

东3局0本场
自摸8s
我打的9m，但是ai建议打9p
![image](https://github.com/chucklu/paipu/assets/3004057/cec93bd5-4807-4a59-bcf5-077819f96cf0)

https://tenhou.net/2/?q=4569m1222569p258s
4569m1222569p258s
打9p 摸[7m8m9m1p2p3p4p7p1s2s3s4s5s6s7s8s9s 60枚]
打9m 摸[1p2p3p4p7p8p9p1s2s3s4s5s6s7s8s9s 56枚]

天凤显示两种打法的进张数差异是有区别，但是肉眼很难一下子看出来，差异在哪里。如果能像我前面说的，把两种差异的结果分别diff一下，并且显示给用户的话，会好很多

**Comment by chucklu** [comment]
再举个例子
https://tenhou.net/2/?q=456m1222569p258s7p
打2s 摸[1p2p3p4p7p8p9p3s4s5s6s7s8s9s 48枚]
打9p 摸[1p2p3p1s2s3s4s5s6s7s8s9s 41枚]
我打的9p，但是ai建议是2s
![image](https://github.com/Equim-chan/mjai-reviewer/assets/3004057/4bbc7961-92b8-4de8-9a95-9d315d8e27a2)

**Comment by Equim-chan** [comment]
在嵌入的天凤牌谱回放播放器里点击手牌能直接转到天凤牌理的对应页面。

如之前所说，目前页面比较混乱，暂时不太想加新的东西。不过之后可能会考虑加入进张数、何切模拟器的信息。

**Comment by Sucaiking** [comment]
在现在的killerducky(beta)界面下，有足够的空间显示何切以及进张期望。
希望项目维护者愿意增加此功能。

**Reply by Equim-chan** [reply]
你可以在 https://github.com/killerducky/killer_mortal_gui/issues 提一个。
